### PR TITLE
fix: always close file handles to stdout/stderr logs

### LINKED ISSUE #260: Error "operation not permitted, unlink chrome-err.log"
#### Environment Information
System: Windows 10
Node.js version: v14.18.1
Chrome version: 96.0.4664.55

#### Provide the steps to reproduce
When I use custom userDataDir, and remove it after chrome exit.

```
const customUserDataDir = 'D:\\.browser-temp\\chrome-4f32efbed74f445c';

const chrome = await launch({
  userDataDir: customUserDataDir
});

chrome.process.on('close', () => {
  if (fs.existsSync(customUserDataDir)) {
    const rm = fs.rmSync || fs.rmdirSync;
  
    rm(customUserDataDir)
  }
});
```

#### What is the current behavior?
```
(node:3204) UnhandledPromiseRejectionWarning: Error: EPERM: operation not permitted, unlink 'D:\\.browser-temp\\chrome-4f32efbed74f445c\\chrome-err.log'
```

#### What is the expected behavior?
```
The outFile and errFile should be handled by fs.closeSync when chrome exit.
```

### PR DESCRIPTION
**Pull request by Junyan**
It should closeSync outFile and errFile when userDataDir is custom.

fixes #260

### CODE REVIEW

File: src/chrome-launcher.ts [hunk 1]
```diff
@@ -405,11 +405,6 @@ class Launcher {
...
   destroyTmp() {
     return new Promise<void>(resolve => {
-      // Only clean up the tmp dir if we created it.
-      if (this.userDataDir === undefined || this.opts.userDataDir !== undefined) {
-        return resolve();
-      }
-
       if (this.outFile) {
         this.fs.closeSync(this.outFile);
         delete this.outFile;
```
Comment (connorjclark) [inline src/chrome-launcher.ts:410]: Makes sense we should always close these file handles when done. I'm curious though, what was the error you were seeing?

**Review by connorjclark — APPROVED** [review]

**Comment by connorjclark** [comment]
Can you run `yarn format`, or allow upstream to push to your fork branches so I can run it?

**Comment by Junyan** [comment]
@connorjclark done

**Comment by kiqaps** [comment]
guys, this still let error file open when used custom data dir, no?

i had to create a patch-package which moves the check condition (if we are using custom data dir) bellow the closeSync of errFile and it worked...

**Pull request merged**